main.cc

```diff
@@ -8,7 +8,15 @@ namespace mold {
 
 std::string_view errno_string() {
   static thread_local char buf[200];
-  strerror_r(errno, buf, sizeof(buf));
+
+  // Special handling for the GNU version of strerror_r(): the function returns
+  // a char * and may completely ignore buf.
+  // The POSIX.1-2001-compliant version of strerror_r() returns an int and
+  // always writes to buf.
+  auto res = strerror_r(errno, buf, sizeof(buf));
```

Comment:
We don't define `_GNU_SOURCE`, so is this really needed?

Reply:
GNU_SOURCE [is on by default](https://godbolt.org/z/P1zfexTTj) in C++ mode. [It's a known issue.](https://gcc.gnu.org/onlinedocs/libstdc++/faq.html#faq.predefined)

Reply:
Changed the implementation to preprocessor fiddling, because the compiler does perform type checking even in the body of an `if constexpr` that evaluates to `false`. I thought I was being clever...

Reply:
I actually like your new code which uses the C preprocessor directive!